sechub-commons-model/src/test/java/com/mercedesbenz/sechub/commons/model/SecHubRemoteCredentialSupportTest.java

```diff
@@ -0,0 +1,181 @@
+package com.mercedesbenz.sechub.commons.model;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.File;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EmptySource;
+import org.junit.jupiter.params.provider.NullSource;
+
+import com.mercedesbenz.sechub.test.TestFileReader;
+
+class SecHubRemoteCredentialSupportTest {
+
+    private SecHubRemoteCredentialSupport supportToTest;
+
+    @BeforeEach
+    void beforeEach() {
+        supportToTest = new SecHubRemoteCredentialSupport();
+    }
+
+    @Test
+    void read_simple_credential_configuration_from_json_string() {
+        /* prepare */
+        String json = TestFileReader.loadTextFile(new File("./src/test/resources/sechub_remote_credentials_simple_config.json"));
+
+        /* execute */
+        SecHubRemoteCredentialConfiguration configuration = supportToTest.getRemoteCredentialConfigurationFromJSONString(json);
+
+        /* test */
+        List<SecHubRemoteCredentialContainer> credentials = configuration.getCredentials();
+        assertEquals(1, credentials.size());
+
+        SecHubRemoteCredentialContainer container = credentials.get(0);
+        assertNotNull(container);
+        assertEquals("example-user", container.getUser());
+        assertEquals("example-api-token-or-password", container.getPassword());
+        assertEquals(".*github.com/*", container.getRemotePattern());
+        assertEquals("example-type", container.getType());
+    }
+
+    @Test
+    void read_empty_credential_configuration_from_json_string() {
+        /* prepare */
+        String json = "{}";
+
+        /* execute */
+        SecHubRemoteCredentialConfiguration configuration = supportToTest.getRemoteCredentialConfigurationFromJSONString(json);
+
+        /* test */
+        assertEquals(0, configuration.getCredentials().size());
+    }
+
+    @Test
+    void read_incomplete_credential_configuration_and_match_location_and_type_where_possible() {
+        /* prepare */
+        String json = TestFileReader.loadTextFile(new File("./src/test/resources/sechub_remote_credentials_incomplete_config.json"));
```

Comment:
As mentioned below, here a text block instead of using json would be easier to understand.
The "incomplete" part is not clear.

When using
```java
String json = """
{
  "credentials": [
    {
      "user" : "example-user",
      "password" : "example-api-token-or-password",
      "remotePattern" : ".*github.com.example*",
      "type" : "git"
    },
    {
      "user" : "user2",
      "password" : "password2",
      "type" : "git"
    },
    {
      "user" : "userX",
      "password" : "tokenX",
      "remotePattern" : ".*example*"
      // type is missing here - meaning "incomplete"
    }
  ]
}
The incomplete part comes visible (I think our jackson Json setup accepts comments, so I used it here)
""";
```